sdk/appconfiguration/app-configuration/review/app-configuration.api.md

```diff
@@ -4,122 +4,74 @@
+    setConfigurationSetting(configurationSetting: SetConfigurationSettingParam, options?: SetConfigurationSettingOptions): Promise<SetConfigurationSettingResponse>;
+    setReadOnly(id: ConfigurationSettingId, options?: SetReadOnlyOptions): Promise<SetReadOnlyResponse>;
+    }
 
 // @public
-export interface AppConfigurationGetRevisionsOptionalParams extends coreHttp.RequestOptionsBase {
-    acceptDatetime?: string;
-    after?: string;
-    key?: string;
-    label?: string;
-    select?: string[];
+export interface ClearReadOnlyOptions extends HttpConditionalFields, RequestOptionsBase {
 }
 
-// @public
-export interface AppConfigurationPutKeyValueOptionalParams extends coreHttp.RequestOptionsBase {
-    entity?: ConfigurationSetting;
-    ifMatch?: string;
-    ifNoneMatch?: string;
-    label?: string;
+// @public (undocumented)
+export interface ClearReadOnlyResponse extends ConfigurationSettingResponse<DeleteLockHeaders> {
 }
 
 // @public
-export interface AppConfigurationPutLockOptionalParams extends coreHttp.RequestOptionsBase {
-    ifMatch?: string;
-    ifNoneMatch?: string;
-    label?: string;
+export interface ConfigurationSetting extends ConfigurationSettingParam {
+    etag?: string;
+    lastModified?: Date;
+    locked?: boolean;
 }
 
 // @public
-export interface ClearReadOnlyOptions extends Pick<AppConfigurationDeleteLockOptionalParams, Exclude<keyof AppConfigurationDeleteLockOptionalParams, 'label'>> {
+export interface ConfigurationSettingId {
+    key: string;
+    label?: string;
 }
 
 // @public
-export interface ConfigurationSetting {
-    // (undocumented)
+export interface ConfigurationSettingParam extends ConfigurationSettingId {
     contentType?: string;
-    // (undocumented)
-    etag?: string;
-    // (undocumented)
-    key: string;
-    // (undocumented)
-    label?: string;
-    // (undocumented)
-    lastModified?: Date;
-    // (undocumented)
-    locked?: boolean;
-    // (undocumented)
     tags?: {
         [propertyName: string]: string;
     };
-    // (undocumented)
     value?: string;
 }
 
-// @public
-export interface ConfigurationSettingId extends Pick<ConfigurationSetting, 'key' | 'label'> {
-}
+// @public (undocumented)
+export type ConfigurationSettingResponse<HeadersT> = ConfigurationSetting & HttpResponseField<HeadersT> & Pick<HeadersT, Exclude<keyof HeadersT, "eTag">>;
 
-// @public
-export interface ConfigurationSettingOptions extends Pick<AppConfigurationPutKeyValueOptionalParams, Exclude<keyof AppConfigurationPutKeyValueOptionalParams, "label" | "entity">> {
+// @public (undocumented)
+export interface DeleteConfigurationSettingOptions extends HttpConditionalFields, RequestOptionsBase {
 }
 
 // @public
-export interface ConfigurationSettingParam extends Pick<ConfigurationSetting, Exclude<keyof ConfigurationSetting, "locked" | "etag" | "lastModified">> {
+export interface DeleteConfigurationSettingResponse extends ConfigurationSettingResponse<DeleteKeyValueHeaders> {
```

Comment:
Whoa, typedoc can render this properly? (Impressed since `ConfigurationSettingResponse` is a type alias.

Reply:
YES IT CA...Oh, dang it. No. I'll flatten some more.

Reply:
Fully flattened. It turns out eliminating the eTag field (in favor of the one already on the object) means that the only field left is syncToken.